pkg/search/postgres/query/uuid_query.go

```diff
@@ -0,0 +1,108 @@
+package pgsearch
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	pkgSearch "github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+func newUUIDQuery(ctx *queryAndFieldContext) (*QueryEntry, error) {
+	whereClause, err := newUUIDQueryWhereClause(ctx.qualifiedColumnName, ctx.value, ctx.queryModifiers...)
+	if err != nil {
+		return nil, err
+	}
+	return qeWithSelectFieldIfNeeded(ctx, &whereClause, nil), nil
+}
+
+func newUUIDQueryWhereClause(columnName string, value string, queryModifiers ...pkgSearch.QueryModifier) (WhereClause, error) {
+	if len(value) == 0 {
+		return WhereClause{}, errors.New("value in search query cannot be empty")
+	}
+
+	if value == pkgSearch.WildcardString {
+		return WhereClause{
+			Query:  fmt.Sprintf("%s is not null", columnName),
+			Values: []interface{}{},
+			equivalentGoFunc: func(foundValue interface{}) bool {
+				foundVal := strings.ToLower(foundValue.(string))
+				return foundVal != ""
+			},
+		}, nil
+	}
+
+	if len(queryModifiers) == 0 {
+		uuidVal, err := uuid.FromString(value)
+		if err != nil {
+			return WhereClause{}, errors.Wrapf(err, "value %q in search query must be valid UUID", value)
+		}
+		return WhereClause{
+			Query:  fmt.Sprintf("%s = $$", columnName),
+			Values: []interface{}{uuidVal},
+			equivalentGoFunc: func(foundValue interface{}) bool {
+				return strings.EqualFold(foundValue.(string), value)
+			},
+		}, nil
+	}
+	if queryModifiers[0] == pkgSearch.AtLeastOne {
+		panic("I dont think this is used")
+	}
```

Comment:
Should we handle this more graceful instead of this panic and return an error here?

I'm not sure whether this is a pattern within the postgres code, I certainly know it's not for the rest of the stackrox code 😛